crates/pipeline_manager/src/db.rs

```diff
@@ -242,57 +249,90 @@ impl ProjectDB {
         Self::connect_inner(connection_str, PoolOptions::new(), initial_sql).await
     }
 
+    fn auto_increment(db_type: &DbType) -> &str {
+        match db_type {
+            DbType::Sqlite => "INTEGER PRIMARY KEY AUTOINCREMENT",
+            DbType::Postgres => "SERIAL PRIMARY KEY",
+        }
+    }
+
+    fn current_time(&self) -> &str {
+        match self.db_type {
+            DbType::Sqlite => "unixepoch('now')",
+            DbType::Postgres => "extract(epoch from now())",
+        }
+    }
+
     /// Connect to the project database.
     async fn connect_inner(
         connection_str: &str,
         pool_options: PoolOptions<Any>,
         initial_sql: &Option<String>,
     ) -> AnyResult<Self> {
+        let db_type = if connection_str.starts_with("sqlite") {
+            DbType::Sqlite
+        } else if connection_str.starts_with("postgres") {
+            DbType::Postgres
+        } else {
+            panic!("Unsupported connection string {}", connection_str)
+        };
         sqlx::any::install_default_drivers();
         debug!("Opening connection to {:?}", connection_str);
         let options =
             AnyConnectOptions::from_str(connection_str)?.log_statements(log::LevelFilter::Trace);
         let conn = pool_options.connect_with(options).await?;
         sqlx::query(
-            r#"
+            format!(
+                "
         CREATE TABLE IF NOT EXISTS project (
-            id integer PRIMARY KEY AUTOINCREMENT,
+            id {},
             version integer NOT NULL,
-            name varchar UNIQUE NOT NULL,
-            description varchar NOT NULL,
-            code varchar NOT NULL,
-            schema varchar NOT NULL,
-            status varchar NOT NULL,
-            error varchar NOT NULL,
-            status_since integer NOT NULL)"#,
+            name text UNIQUE NOT NULL,
```

Comment:
I think varchar works if we're not using the 0.7.0alpha and instead use 0.6 for sqlx